agent/src/test/java/com/teamscale/jacoco/agent/store/upload/delay/DelayedCommitDescriptorRetrievalTest.java

```diff
@@ -16,23 +18,25 @@
 	@Test
 	public void locatorShouldTriggerUploadOfCachedXmls() throws Exception {
 		ExecutorService storeExecutor = Executors.newSingleThreadExecutor();
-		InMemoryStore cache = new InMemoryStore();
-		InMemoryStore destination = new InMemoryStore();
-		DelayedCommitDescriptorStore store = new DelayedCommitDescriptorStore(commit -> destination, cache,
+		File coverageFile = File.createTempFile("jacoco-", ".xml");
+		Path outputPath = coverageFile.getParentFile().toPath();
+
+		InMemoryUploader destination = new InMemoryUploader();
+		DelayedCommitDescriptorStore store = new DelayedCommitDescriptorStore(commit -> destination, outputPath,
 				storeExecutor);
 
 		ExecutorService locatorExecutor = Executors.newSingleThreadExecutor();
 		GitPropertiesLocator locator = new GitPropertiesLocator(store, locatorExecutor);
 
-		store.store("xml1");
+		store.upload(coverageFile);
 		locator.searchJarFileForGitPropertiesAsync(new File(getClass().getResource("git-properties.jar").toURI()));
 		locatorExecutor.shutdown();
 		locatorExecutor.awaitTermination(5, TimeUnit.SECONDS);
 		storeExecutor.shutdown();
 		storeExecutor.awaitTermination(5, TimeUnit.SECONDS);
 
-		assertThat(cache.getXmls()).isEmpty();
-		assertThat(destination.getXmls()).containsExactly("xml1");
+		assertThat(Files.list(outputPath).anyMatch(path -> path.getFileName().equals(coverageFile.toPath())))
+				.isFalse();
```

Comment:
please convert the stream to a list and use proper list assertions, as these give nicer error masseges than `isFalse`